.github/workflows/docs.yaml

```diff
@@ -12,9 +12,6 @@ concurrency:
   group: ${{ github.workflow }}-${{ github.ref }}
   cancel-in-progress: true
 
-permissions:
-  contents: read
```

Comment:
oh wrong permissions; this is required/suggested by OpenSSF Scorecard as least privilege on all top-level GH Workflows. (As one should never override the top-level, only on a per job level)

I meant lines 18-19 below this can be removed, the `write` permissions that were specifically used for the `docs` GH Job, which was only needed to write to the `gh-pages` branch.

Summarizing, the `read` permissions here should be left in, the `write` permissions below should be removed.

Reply:
@agilgur5 I'll put the read permissions back.  Wont' removing the `write` permissions cause the `Upload Docs Site` step to fail?  [Its README](https://github.com/actions/upload-artifact?tab=readme-ov-file) doesn't make me think so but was curious.  Putting `read` back and removing `write`.

Reply:
No, `actions/upload-artifact` is CI/GHA specific; i.e. it's writing an artifact to GHA which actually doesn't need extra permissions; similar to how writing to the GHA cache doesn't.

`contents: write` [specifically](https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs) is the ability to write to the repo _itself_ (GHA's artifact server is independent of that)